Fix ReplaceExprsInScope

tlemo opened the pull request:
Fix for issue #88

tlemo committed:
Fix ReplaceExprsInScope

csarofeen commented on torch/csrc/jit/codegen/cuda/lower_utils.cpp:
```diff
@@ -200,56 +200,45 @@ struct CloneLoopNest : public OptOutMutator {
+ private:
+  explicit ReplaceExprsInScope(
+      const std::unordered_map<Expr*, Expr*>& replacement_map)
+      : replacement_map_(replacement_map) {}
```
clang-tidy decrees std::move. All praise clang-tidy.

csarofeen reviewed (approved):
Just make sure to fix clang-tidy.

tlemo committed:
Please clang-tidy

... although the clang-tidy recommendation is very questionable
(moves are cheaper than copies, but not free)
